### DISCUSSION TITLE
Perform a natural language search for a function within the local code repository.

### DISCUSSION THREAD

**Discussion by yakiler** [community/community#56289] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Using natural language, search for an already implemented function. For example, if I want to check if there is a function for email validation in my code repository, traditional searches would involve repeatedly using various keywords to perform fuzzy queries. Now that Copilot has access to all my code, can I use 'search for email validation feature or function' to retrieve code from my local repository?

**Comment by look** [comment, marked as answer]
Definitely a cool idea @yakiler! This is something we have discussed, but requires quite a lot of work to implement at scale.
